addons/website_customer/controllers/main.py

```diff
@@ -15,47 +15,62 @@ class WebsiteCustomer(http.Controller):
     @http.route([
         '/customers',
         '/customers/page/<int:page>',
-        '/customers/country/<int:country_id>',
-        '/customers/country/<country_name>-<int:country_id>',
-        '/customers/country/<int:country_id>/page/<int:page>',
-        '/customers/country/<country_name>-<int:country_id>/page/<int:page>',
-        '/customers/tag/<tag_id>',
-        '/customers/tag/<tag_id>/page/<int:page>',
-        '/customers/tag/<tag_id>/country/<int:country_id>',
-        '/customers/tag/<tag_id>/country/<country_name>-<int:country_id>',
-        '/customers/tag/<tag_id>/country/<int:country_id>/page/<int:page>',
-        '/customers/tag/<tag_id>/country/<country_name>-<int:country_id>/page/<int:page>',
+        '/customers/country/<model("res.country"):country>',
+        '/customers/country/<model("res.country"):country>/page/<int:page>',
+        '/customers/industry/<model("res.partner.industry"):industry>',
+        '/customers/industry/<model("res.partner.industry"):industry>/page/<int:page>',
+        '/customers/industry/<model("res.partner.industry"):industry>/country/<model("res.country"):country>',
+        '/customers/industry/<model("res.partner.industry"):industry>/country/<model("res.country"):country>/page/<int:page>',
     ], type='http', auth="public", website=True)
-    def customers(self, country_id=0, page=0, country_name='', tag_id=0, **post):
-        Country = request.env['res.country']
+    def customers(self, country=None, industry=None, page=0, **post):
         Tag = request.env['res.partner.tag']
         Partner = request.env['res.partner']
         partner_name = post.get('search', '')
 
         domain = [('website_published', '=', True), ('assigned_partner_id', '!=', False)]
         if partner_name:
             domain += [
-                '|',
+                '|', '|',
                 ('name', 'ilike', post.get("search")),
-                ('website_description', 'ilike', post.get("search"))
+                ('website_description', 'ilike', post.get("search")),
+                ('industry_id.name', 'ilike', post.get("search")),
```

Comment:
`post.get("search")` is `partner_name`  (probably deserves a better variable name, though)